src/org/pentaho/reporting/platform/plugin/output/CachingPageableHTMLOutput.java

```diff
@@ -241,7 +242,7 @@ IReportContent regenerateCache( final MasterReport report, final int yieldRate,
     throws ReportProcessingException {
     logger.warn( "Regenerating report data for " + key );
     final IReportContent result = produceCacheablePages( report, yieldRate, key, acceptedPage );
-    persistContent( key, result );
+    persistContent( key, result, null );
```

Comment:
This will backfire. You are storing content without a totalRows indicator, so the next time you retrieve it from the cache, you are missing that information. 

The "produceCachablePages" method will update the listener state from the events that get fired during the report processing, but without persisting that information, you are producing an invalid state.

Create a unit-test where you can replay that scenario (retrieve from cache, but not using the "listener.isFirstPageMode") and check what your cache contains. You will find that you have no totalRows stored and that subsequent cache-retrievals will therefore not set the correct totalRows counter.